[quant][fx] Fix weight_dtype and bias_dtype backend_config checks

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86719

Summary:
This PR adds checks for the existence of "weight_dtype" and "bias_dtype" in the node_name_to_dtype dictionary before accessing it,
the corner case is hit when we check the compatibility of qconfig and backend_config for weight and bias that appears before activation (e.g. torch.addmm)

Test Plan:
python test/test_quantization.py -k test_backend_config_check_for_weight_and_bias

Reviewers:

Subscribers:

Tasks:

Tags:

jerryzh168 committed:
[quant][fx] Fix weight_dtype and bias_dtype backend_config checks

Summary:
This PR adds checks for the existence of "weight_dtype" and "bias_dtype" in the node_name_to_dtype dictionary before accessing it,
the corner case is hit when we check the compatibility of qconfig and backend_config for weight and bias that appears before activation (e.g. torch.addmm)

Test Plan:
python test/test_quantization.py -k test_backend_config_check_for_weight_and_bias

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

andrewor14 commented on torch/ao/quantization/fx/prepare.py:
```diff
@@ -195,14 +195,18 @@ def is_input_arg_dtype_supported_by_backend(
         return bias_dtype is None or \
-            node_name_to_target_dtype[node.name]["bias_dtype"][0] == bias_dtype  # type: ignore[index]
+            (
+                "bias_dtype" in node_name_to_target_dtype and
```
I'm a bit confused about what this means. Should this be `node_name_to_target_dtype[node.name]`? I don't think we ever set `node_name_to_target_dtype["bias_dtype"]`

jerryzh168 replied:
oh right, will fix

jerryzh168 committed:
Update on "[quant][fx] Fix weight_dtype and bias_dtype backend_config checks"


Summary:
This PR adds checks for the existence of "weight_dtype" and "bias_dtype" in the node_name_to_dtype dictionary before accessing it,
the corner case is hit when we check the compatibility of qconfig and backend_config for weight and bias that appears before activation (e.g. torch.addmm)

Test Plan:
python test/test_quantization.py -k test_backend_config_check_for_weight_and_bias

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@pytorchbot merge -g
